core/src/test/java/org/elasticsearch/common/lucene/search/MultiPhrasePrefixQueryTests.java

```diff
@@ -63,4 +62,35 @@ public void simpleTests() throws Exception {
         query.add(new Term("field", "xxx"));
         assertThat(Lucene.count(searcher, query), equalTo(0l));
     }
+
+    @Test
+    public void testBoost() throws Exception {
+        IndexWriter writer = new IndexWriter(new RAMDirectory(), new IndexWriterConfig(Lucene.STANDARD_ANALYZER));
+        Document doc = new Document();
+        doc.add(new Field("field", "aaa bbb", TextField.TYPE_NOT_STORED));
+        writer.addDocument(doc);
+        doc = new Document();
+        doc.add(new Field("field", "ccc ddd", TextField.TYPE_NOT_STORED));
+        writer.addDocument(doc);
+        IndexReader reader = DirectoryReader.open(writer, true);
+        IndexSearcher searcher = new IndexSearcher(reader);
+
+        BooleanQuery booleanQuery = new BooleanQuery();
+        TermQuery termQuery = new TermQuery(new Term("field", "aaa"));
+        termQuery.setBoost(10);
+        booleanQuery.add(new BooleanClause(termQuery, BooleanClause.Occur.SHOULD));
+        MultiPhrasePrefixQuery multiPhrasePrefixQuery = new MultiPhrasePrefixQuery();
+        multiPhrasePrefixQuery.add(new Term[]{new Term("field", "ccc"), new Term("field", "dd")});
+        booleanQuery.add(new BooleanClause(multiPhrasePrefixQuery, BooleanClause.Occur.SHOULD));
+        TopDocs topDocs = searcher.search(booleanQuery, 2);
+        assertThat(topDocs.totalHits, equalTo(2));
+        assertThat(topDocs.scoreDocs[0].doc, equalTo(0));
+        assertThat(topDocs.scoreDocs[1].doc, equalTo(1));
+
+        multiPhrasePrefixQuery.setBoost(100);
+        topDocs = searcher.search(booleanQuery, 2);
+        assertThat(topDocs.totalHits, equalTo(2));
+        assertThat(topDocs.scoreDocs[0].doc, equalTo(1));
+        assertThat(topDocs.scoreDocs[1].doc, equalTo(0));
+    }
```

Comment:
Can we make the test simpler and just call Query.rewrite and ensure the boost is preserved?